lib/src/op_heads_store.rs

```diff
@@ -66,16 +82,20 @@ pub fn resolve_op_heads<E>(
     resolver: impl FnOnce(Vec<Operation>) -> Result<Operation, E>,
 ) -> Result<Operation, E>
 where
-    E: From<OpHeadResolutionError> + From<OpStoreError>,
+    E: From<OpHeadResolutionError> + From<OpHeadsStoreError> + From<OpStoreError>,
 {
     // This can be empty if the OpHeadsStore doesn't support atomic updates.
     // For example, all entries ahead of a readdir() pointer could be deleted by
     // another concurrent process.
-    let mut op_heads = op_heads_store.get_op_heads();
+    let mut op_heads = op_heads_store
+        .get_op_heads()
+        .map_err(|err| OpHeadsStoreError::Read(err.into()))?;
```

Comment:
just `?`? Doesn't it wrap `OpHeadsStoreError` twice?
(applies to other error propagation paths)